cmd/tempo-vulture/main.go

```diff
@@ -137,27 +144,72 @@ func main() {
 				continue
 			}
 
-			r := newRand(seed)
-			hexID := fmt.Sprintf("%016x%016x", r.Int63(), r.Int63())
+			log := logger.With(
+				zap.String("org_id", tempoOrgID),
+			)
+
+			client := util.NewClient(tempoQueryURL, tempoOrgID, log)
 
 			// query the trace
-			metrics, err := queryTempoAndAnalyze(tempoQueryURL, seed, hexID)
+			queryMetrics, err := queryTrace(client, seed)
 			if err != nil {
 				metricErrorTotal.Inc()
+				log.Error("query for metrics failed",
+					zap.Error(err),
+				)
 			}
+			pushMetrics(queryMetrics)
+		}
+	}()
+
+	// Search
+	go func() {
```

Comment:
Should we just smoosh the readers together?  We could still keep them on a different ticker, but just running out of the same routine.

Reply:
I think the way you have it is fine.